Realm/RLMDictionary.mm

```diff
@@ -493,6 +493,22 @@ - (id)objectAtIndex:(NSUInteger)index {
     @throw RLMException(@"This method is not available on RLMDictionary.");
 }
 
+- (nullable id)firstObject {
```

Comment:
Why not just mark these as `@optional`?

Reply:
It maybe should be. This is weirdness resulting from RLMDictionary conforming to RLMCollection but Map not conforming to RealmCollection. Everything which conforms to RealmCollection has first/last so it's sort of awkward to have to handle those being optional on the RLMCollection.

Reply:
I've switched the ones which previously were optional back to being optional.